pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -230,6 +230,52 @@ func (a Adapter) waitForManifestDeployCompletion(applicationName string, gvr sch
 	}
 }
 
+// GetProtocol returns the protocol string
+func getProtocol(route routev1.Route) string {
+	if route.Spec.TLS != nil {
+		return "https"
+	}
+	return "http"
+}
+
+func (a Adapter) getApplicationURL(applicationName string) (fullURL string, err error) {
+
+	// TODO: Can we use a occlient created somewhere else rather than create another
+	client, err := occlient.New()
+	if err != nil {
+		return "", err
+	}
+
+	// Need to wait for a second to give the server time to create the artifacts
+	// TODO: Replace wait with a wait for object to be created (need to determine which object!!!)
+	time.Sleep(2 * time.Second)
+
+	fullURL = ""
+	routeSupported, err := client.IsRouteSupported()
+	if routeSupported {
+		route, err := client.GetRoute(applicationName)
```

Comment:
@EnriqueL8 Would you have any issues changing this back to a .List function rather than a .Get.
Reason is that when using an Ingress on OCP it automatically generates a Route (via the OCP Ingress operator) with a generated name, ie nodejs-deploy-j4d9n and the .get for "nodejs-deploy" will not find it. If we use .List we will check for the label "component=nodejs-deply" (which you and Luke added) and which should only return 1  object, which would then be used. Thoughts?

Reply:
yup, I just checked: `func (c *Client) ListRoutes(labelSelector string) ([]routev1.Route, error)` so that should only return one route in our case

Reply:
Changed